src/ansible_compat/runtime.py

```diff
@@ -285,15 +285,19 @@ def load_collections(self) -> None:
             _logger.error(proc)
             msg = f"Unable to list collections: {proc}"
             raise RuntimeError(msg)
-        data = json.loads(proc.stdout)
+        try:
+            data = json.loads(proc.stdout)
+        except json.decoder.JSONDecodeError as exc:
+            msg = f"Unable to parse galaxy output as JSON: {proc.stdout}"
+            raise RuntimeError(msg) from exc
         if not isinstance(data, dict):
             msg = f"Unexpected collection data, {data}"
             raise TypeError(msg)
         for path in data:
+            if not isinstance(data[path], dict):
+                msg = f"Unexpected collection data, {data[path]}"
+                raise TypeError(msg)
             for collection, collection_info in data[path].items():
-                if not isinstance(collection, str):
```

Comment:
This is removed as I don't think it's possible to hit. Non-string keys won't get this far, as they won't be recognized as valid JSON and therefore hit the DecodeError above.